tests/test_api.py

```diff
@@ -84,4 +91,118 @@ def test_qasm_to_phir(self) -> None:
         measure q[1]->cr[0];
         """
 
-        assert qasm_to_phir(qasm, QtmMachine.H1_1)
+        phir = qasm_to_phir(qasm, QtmMachine.H1_1)
+        PHIRModel.model_validate_json(phir)
+
+    def test_qasm_to_phir_with_wasm(self) -> None:
+        """Test the qasm string entrypoint works with WASM."""
+        qasm = """
+        OPENQASM 2.0;
+        include "qelib1.inc";
+
+        qreg q[2];
+        h q;
+        ZZ q[1],q[0];
+        creg cr[3];
+        creg cs[3];
+        creg co[3];
+        measure q[0]->cr[0];
+        measure q[1]->cr[1];
+
+        cs = cr;
+        co = add(cr, cs);
+        """
+
+        wasm_bytes = get_wat_as_wasm_bytes(WatFile.add)
+
+        phir_str = qasm_to_phir(qasm, QtmMachine.H1_1, wasm_bytes=wasm_bytes)
+        phir = json.loads(phir_str)
+        assert phir is not None
+
+        expected_metadata = {
+            "ff_object": (
+                "WASM module uid: 28c0194b91f1e24d6fc40ec480c026a5874661184"
+                "bcd411a61bd5d5383df5180"
```

Comment:
Can we document somewhere how these module uid's are generated? They seem magic to me.

Reply:
I actually don't know how other than to look after the fact in the PHIR.

Reply:
I think I'd like to find out. Or perhaps we should not assert for module id if it's irrelevant.